arangod/Aql/QueryCursor.h

```diff
@@ -95,13 +95,17 @@ class QueryResultCursor final : public arangodb::Cursor {
 /// Cursor managing a query from which it continuously gets
 /// new results. Query, transaction and locks will live until
 /// cursor is deleted (or query exhausted)
-class QueryStreamCursor final : public arangodb::Cursor {
+class QueryStreamCursor final : public Cursor {
+  struct Token {};
+
  public:
-  QueryStreamCursor(std::shared_ptr<aql::Query> q, size_t batchSize, double ttl,
-                    bool isRetriable,
-                    transaction::OperationOrigin operationOrigin);
+  static auto create(std::shared_ptr<Query> q, size_t batchSize, double ttl,
+                     bool isRetriable)
+      -> futures::Future<std::unique_ptr<QueryStreamCursor>>;
 
-  ~QueryStreamCursor();
+  QueryStreamCursor(Token, std::shared_ptr<Query> q, size_t batchSize,
```

Comment:
It seems that there is only one constructor of QueryStreamCursor, even after this change.
But this PR adds the `Token` to the constructor call and also adds a public static `create` method. 
I am not clear why we have all this now. It would make more sense to me to remove `Token` again, and make the constructor of `QueryStreamCursor` private, so that it can only be created via the static `create` method.

Reply:
`QueryStreamCursor` is constructed via `make_unique`, which won't work if the constructor is private. The Token is to mitigate that: As it itself is private, other users can't create one, and without it can't call the constructor, making it effectively private.

Reply:
Ok